src/auth/ldap_krb/KrbClientHandler.cpp

```diff
@@ -0,0 +1,279 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 Daniel Oliveira <doliveira@suse.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+/* Include order and names:
+ * a) Immediate related header
+ * b) C libraries (if any),
+ * c) C++ libraries,
+ * d) Other support libraries
+ * e) Other project's support libraries
+ *
+ * Within each section the includes should
+ * be ordered alphabetically.
+ */
+
+#include "KrbClientHandler.hpp"
+
+#include <errno.h>
+#include <string>
+#include "KrbProtocol.hpp"
+
+#include "auth/KeyRing.h"
+#include "common/config.h"
+#include "common/dout.h"
+
+#define dout_subsys ceph_subsys_auth
+#undef dout_prefix
+#define dout_prefix *_dout << "krb5/gssapi request: "
+
+struct AuthAuthorizer;
+
+template<LockPolicy lp> 
+AuthAuthorizer* KrbClientHandler<lp>::
+build_authorizer(uint32_t service_id) const 
+{
+  constexpr auto SUBSYSTEM_ID(20);
+
+  ldout(cct, SUBSYSTEM_ID) 
+      << "KrbClientHandler::build_authorizer(): Service: " 
+      << ceph_entity_type_name(service_id) << dendl; 
+
+  KrbAuthorizer* krb_auth = new KrbAuthorizer();
+  if (krb_auth) {
+    krb_auth->build_authorizer(cct->_conf->name, global_id);
+  }
+  return krb_auth;
+}
+
+template<LockPolicy lp> 
+KrbClientHandler<lp>::~KrbClientHandler() 
+{
+  OM_uint32 gss_major_status(0); 
+  OM_uint32 gss_minor_status(0); 
+
+  gss_release_name(&gss_minor_status, &m_gss_client_name);
+  gss_release_name(&gss_minor_status, &m_gss_service_name);
+  gss_release_cred(&gss_minor_status, &m_gss_credentials); 
+  gss_delete_sec_context(&gss_minor_status, &m_gss_sec_ctx, GSS_C_NO_BUFFER); 
+  gss_release_buffer(&gss_minor_status, static_cast<gss_buffer_t>(&m_gss_buffer_out)); 
+}
+
+template<LockPolicy lp>
+int KrbClientHandler<lp>::build_request(bufferlist& buff_list) const
+{
+  constexpr auto SUBSYSTEM_ID(20);
```

Comment:
this is being used as the log level, not a subsystem id. Why not just explicitly setting the log level on ldout instead?